test/js/calc-length.js

```diff
@@ -83,4 +83,74 @@ suite('CalcLength', function() {
     var dividedCalcLength = calcLength.divide(2.5);
     assert.strictEqual(dividedCalcLength.cssString, 'calc(10px+1.28em)');
   });
+
+  test('Adding two CalcLengths returns a new CalcLength with expected values in each value type', function() {
+    var cLength_1 = new CalcLength({px: 15, em: 6})
+    var cLength_2 = new CalcLength({px: 10, em: 3})
+    var lengthAddition = cLength_1.add(cLength_2);
+    var expectedlengthAddition = new CalcLength({px: 25, em: 9})
+
+    assert.instanceOf(lengthAddition, CalcLength, 'two added CalcLength of same type should be an instance of CalcLength');
+    assert.isTrue(expectedlengthAddition.equals(lengthAddition));
+  });
+
+    test('CalcLength do not have to have the same value types in order to be added and all value types seen in both'
+        +'CalcLengths with be in the outputted CalcLength', function() {
+    var cLength_1 = new CalcLength({px: 15, em: 6, percent: 5})
+    var cLength_2 = new CalcLength({px: 10, em: 3, ex: 6})
+    var lengthAddition = cLength_1.add(cLength_2);
+    var expectedlengthAddition = new CalcLength({px: 25, em: 9, percent: 5, ex: 6});
+
+    assert.instanceOf(lengthAddition, CalcLength, 'two added CalcLength of same type should be an instance of CalcLength');
+    assert.isTrue(expectedlengthAddition.equals(lengthAddition));
+  });
+
+  test('Subtracting two CalcLengths returns a new CalcLength with expected values in each value type', function() {
+    var cLength_1 = new CalcLength({px: 15, em: 6})
+    var cLength_2 = new CalcLength({px: 10, em: 3})
+    var lengthSubtraction = cLength_1.subtract(cLength_2);
+    var expectedlengthSubtraction = new CalcLength({px: 5, em: 3})
+
+    assert.instanceOf(lengthSubtraction, CalcLength, 'two added CalcLength of same type should be an instance of CalcLength');
+    assert.isTrue(expectedlengthSubtraction.equals(lengthSubtraction));
+  });
+
+    test('CalcLength do not have to have the same value types in order to be subtracted from each other and all value types '
```

Comment:
Likewise, how about

"All values in subtracted CalcLengths appear in the output"